### DISCUSSION TITLE
FFT performance boost - Ne10 backend

### DISCUSSION THREAD

**Discussion by pfeatherstone** [davisking/dlib#3030] [Ideas]
In my tests, the Ne10 library has much faster FFTs than the FFTW library on ARM. So it would be good to add this as a backend.

**Comment by pfeatherstone** [comment]
Though it only works for powers of 2

**Comment by davisking** [comment]
Yeah.  You could have a code path for power of two sizes that switches to that :shrug:

**Reply by pfeatherstone** [reply]
Though I might try adding a sprinkle of SIMD on the kissFFT backend at some point to see if I can close the gap on ARM.

**Reply by davisking** [reply]
> Though I might try adding a sprinkle of SIMD on the kissFFT backend at some point to see if I can close the gap on ARM. 

That would be cool. Especially since then it's not another dependency 😁
